src/Catalog/Persistence/AzureStorage.cs

```diff
@@ -283,5 +283,24 @@ public override Uri GetUri(string name)
         {
             return new Uri(_directory.Uri, name);
         }
+
+        public override async Task<bool> IsUpdateRequiredAsync(Uri resourceUri, string id, string version)
```

Comment:
These APIs are meant to be a general purpose storage API--not specific to NuGet or packaging. `AzureStorage` should know nothing about `id`s or `version`s--it should just accept a `resourceUri`. Additionally, by adding package information to these calls, you get issues such as the one that @joelverhagen calls out below where we're duplicating code from `DnxMaker`.

I would suggest rewriting this method to take in just `resourceUri` and rewriting `DnxMaker` to construct the `resourceUri`.

Reply:
Updated to use DnxMarker